### PR TITLE
Use builtin min and max functions

### PR DESCRIPTION
**Pull request by jhalterman**
With the upgrade to Go 1.21 (#7282), grpc-go can use the new builtin [min](https://pkg.go.dev/builtin#min) and [max](https://pkg.go.dev/builtin#max) functions. This commit removes existing functions and uses the built-ins instead.

RELEASE NOTES: none

### CODE REVIEW

**Review by purnesh42H — APPROVED** [review]
lgtm. Thanks for the contribution!

File: internal/transport/controlbuf.go [hunk 1]
```diff
@@ -30,6 +30,7 @@ import (
 
 	"golang.org/x/net/http2"
 	"golang.org/x/net/http2/hpack"
+
```
Comment (arvindbr8) [inline internal/transport/controlbuf.go:33]: revert unrelated changes, pls?

**Review by arvindbr8 — APPROVED** [review]
LGTM, do you mind reverting the unrelated changes please?

File: internal/transport/controlbuf.go (hunk 1 above)
Comment (arvindbr8) [reply internal/transport/controlbuf.go:33]: and others?

Comment (jhalterman) [reply internal/transport/controlbuf.go]: Sure, done!

**Review by dfawley — APPROVED** [review]
